### ISSUE TITLE
Update Cake.Incubator add-in to target Cake v3.0.0

### ISSUE THREAD

**Issue by nils-a** [cake-contrib/Cake.Incubator#216]
This is currently blocked by https://github.com/cake-contrib/Cake.Incubator/pull/215

**Comment by daverogersvald** [comment]
Just a little confused. [On this page](https://cakebuild.net/extensions/cake-incubator/), it states that v 3.0.0 of Cake is supported.

Yet my reading of this issue suggests that it is not. Given that I receive this error, it also suggests that it is not:
`
The assembly 'Cake.Incubator, Version=7.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (2.0.0).
For best compatibility it should target Cake.Core version 3.0.0.
`

Is there a tweak I can make to get around that exception, or is 3.0.0 just not supported?
If so, perhaps someone could edit the page referenced above to update that.

Cheers

**Comment by Dusty-Meg** [comment]
From my understanding of that page, it is not inclusive of version 3.0.0, its all versions up to that.
The code will need to be updated and a new version pushed before cake 3.0.0 will be supported.

**Comment by daverogersvald** [comment]
Really? `Versions: 2.0.0 - 3.0.0` means 3.0.0 is not included?
That seems to be a bit odd.
This would have made more sense to me: `Versions: 2.0.0 < 3.0.0`
But if that is the case, then fair enough.

**Comment by pascalberger** [comment]
> Really? `Versions: 2.0.0 - 3.0.0` means 3.0.0 is not included? 

This was actually a bug on the website. The number is inclusive, but should have been `2.0.0 - 2.3.0`. It has been fixed now.

**Comment by wgnf** [comment]
Upgraded to Cake Version 3.0.0 at work the other day and I'm currently running into issues using `Cake.Incubator` there:

```
Error: Could not load type 'Cake.Common.Tools.DotNetCore.Test.DotNetCoreTestSettings' from assembly 'Cake.Common, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null'.
```

I'm pretty sure this is because there are still some places where classes with the prefix `DotNetCore` are used, which have been removed in [Cake 3.0.0](https://cakebuild.net/blog/2022/11/cake-v3.0.0-released#breaking-changes)

i.e. see the usage of `DotNetCoreTestSettings` here:

- https://github.com/cake-contrib/Cake.Incubator/blob/develop/src/Cake.Incubator/DotNetCoreTestExtensions.cs#L32
- https://github.com/cake-contrib/Cake.Incubator/blob/develop/src/Cake.Incubator/DotNetCoreTestExtensions.cs#L46

idk what you folks are thinking, but I would be up to change all the places where something `DotNetCore*` is used in favor of the `DotNet*` equivalents and open a PR. Should this be discussed in a new issue (i.e. should extensions like `DotNetCoreTestExtensions` also be renamed in `DotNetTestExtensions`, ...)?

**Comment by nils-a** [comment]
> ...  change all the places where something `DotNetCore*` is used in favor of the `DotNet*` equivalents and open a PR. Should this be discussed in a new issue

Someone started out in #215, then closed it after my comments, and wrote they'd open a new pr.
But that was some time ago, so: If you're up to it, feel free to open a PR.

> should extensions like `DotNetCoreTestExtensions` also be renamed in `DotNetTestExtensions` ?

Yes, we probably should have done this along with the deprecation in Cake some time ago. Since the switch to 3.0.0 will be breaking anyway, I'd say we should change the names "here" as well.

Any opinions, regarding the change from `DotNetCore*` "here" to `DotNet*`? 
/cc @gep13, @pascalberger, @augustoproiete

**Comment by wgnf** [comment]
Maybe it would be good to have two separate PRs though. One for fixing the current problem with 3.0.0 compatibility and a second for changing the naming from `DotNetCore*` to `DotNet*` in this Addin.  
  
so that the first PR might be shipped as a fix (?) and second as a Breaking Change when switching over to Cake 3.0.0

**Comment by pascalberger** [comment]
> Any opinions, regarding the change from `DotNetCore*` "here" to `DotNet*`? 

Makes sense to me 👍

**Comment by wgnf** [comment]
I added the PR #221 for my attempt at upgrading to Cake 3.0.0, as I was thinking it doesn't really make sense to just change some bits of the upgrade, when I can just do everything at once. Upside is with the direct upgrade to cake 3.0.0 I was able to see which members have been removed with 3.0.0.  
  
I hope this is alright and does not break any rules/the discussion before.
